File: EEFRT Demo Android/app/src/main/java/ai/a2i2/conductor/effrtdemoandroid/ui/data/CloseMessage.kt
package ai.a2i2.conductor.effrtdemoandroid.ui.data

import kotlinx.serialization.Serializable

@Serializable
data class CloseMessage(
    val shouldShowExitDialog: Boolean,
    val incrementAttemptCount: Boolean,
    val taskRequiresRestart: Boolean,
)

File: public/src/embedContext/CloseMessage.js
export default class CloseMessage {
    constructor(shouldShowExitDialog, incrementAttemptCount, taskRequiresRestart) {
        this.shouldShowExitDialog = shouldShowExitDialog;
        this.incrementAttemptCount = incrementAttemptCount;
        this.taskRequiresRestart = taskRequiresRestart;
    }

    stringify() {
        return JSON.stringify(this);
    }
}
